### PR TITLE
unbind the CheckConsistency for some structs so that they can be moved to sub packages easier

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
I can see `github.com/stretchr/testify/assert` and `t *testing.T` in non-unit-test code.

Should they be moved into unit test code?

File: models/consistency.go [hunk 1]
```diff
@@ -46,17 +42,34 @@ func CheckConsistencyFor(t *testing.T, beansToCheck ...interface{}) {
...
+	case *Milestone:
+		checkForConsistencyMilestone(b, t)
+	case *Label:
+		checkForConsistencyLabel(b, t)
+	case *Team:
+		checkForConsistencyTeam(b, t)
+	case *Action:
+		checkForConsistencyAction(b, t)
+	default:
+		t.Errorf("unknow bean type: %#v", bean)
```
Comment (delvh) [inline models/consistency.go]:
```suggestion
		t.Errorf("unknown bean type: %#v", bean)
```

File: models/consistency.go [hunk 2]
```diff
@@ -70,7 +83,7 @@ func assertCount(t *testing.T, bean interface{}, expected int) {
 		"Failed consistency test, the counted bean (of type %T) was %+v", bean, bean)
 }
 
-func (user *User) checkForConsistency(t *testing.T) {
+func checkForConsistencyUser(user *User, t *testing.T) {
```
Comment (delvh) [inline models/consistency.go]: I'd say that is one of two possible naming schemes.
The other would have been
```suggestion
func checkForUserConsistency(user *User, t *testing.T) {
```
Same below then.
What do you say about this approach?

**Comment by lunny** [comment]
> I can see `github.com/stretchr/testify/assert` and `t *testing.T` in non-unit-test code.
> 
> Should they be moved into unit test code?

I have tried that. Unfortunately, they cannot. I think that's should be a limit of Golang test file. You cannot import a function on a test file.

**Comment by wxiaoguang** [comment]
How about using a separate package for unit test related code, then maybe this package won't be compiled for releases.

**Comment by lunny** [comment]
> How about using a separate package for unit test related code, then maybe this package won't be compiled for releases.

We can't do that. That will result in recycle import. The new package will depend models, but models will use method `CheckConsistencyFor`.

**Review by wxiaoguang — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**